1112: Introduce new map style variant

kadamwhite opened the pull request:
This PR introduces a new side-by-side map style variant, for use on pages where we want the text to be more in-line with the map on desktop viewports.

<img width="1302" alt="image" src="https://github.com/user-attachments/assets/ad4034ca-05f0-4b4f-9f64-4a822ac0525b" />

## Remaining issues

The template for the marker block is a very verbose and previous-annual-report-tailored situation, so **_duplicating_** a slide is currently an easier way to insert a new item into the map

![slide-duplication-flow](https://github.com/user-attachments/assets/00ba423c-57ac-4958-90b9-5304b36819f2)

kadamwhite committed:
1112: Begin introducing new side-by-side desktop map view

kadamwhite committed:
1112: Map currently relies on an ID, set "multiple" to "false"

kadamwhite committed:
SCSS: Resolve mixed-decls deprecation warning

kadamwhite committed:
SCSS: Switch from deprecated @import to @use, where possible

kadamwhite committed:
Merge branch 'main' into 1112-map-style-variant

kadamwhite committed:
1112: Make it obvious that attributes are float values

kadamwhite committed:
1112: Avoid double-initializing the map in DevServer

kadamwhite committed:
1112: Refactor reinitialization guard so it uses a module-external flag

kadamwhite committed:
1112: Remove unnecessary defensive guards on objects we know will exist

kadamwhite committed:
Tech Debt: Switch from IconButton to Button in slider navigation component

kadamwhite committed:
Tech Debt: Adjust import of getBlocksByClientId to resolve deprecation warning

kadamwhite committed:
Tech Debt: Fix unstable reference returned from useSelect which triggers a console warning

kadamwhite committed:
1112: Improve styling and layout for side-by-side variant on frontend and in-editor

kadamwhite committed:
1112: Keep map fully rendered if sidebar height changes

kadamwhite committed:
Format JS

kadamwhite committed:
1112: Correctly resize map when changing slide

joeleenk reviewed (commented):
Looking good overall, just a tiny question, and a couple of testing comments:

- The map and all of it's contents default to full width, and the map block has the `alignfull` class applied, even when it's aligned wide, which causes some layout problems if it's used at wide width at the top level.
- Arrow alignment needs work, which you mentioned already.

kadamwhite commented on src/blocks/map/view.js:
```diff
@@ -371,13 +383,13 @@ document.addEventListener( 'DOMContentLoaded', () => {
 							geometry: {
 								type: 'Point',
 								coordinates: [
-									marker?.dataset?.long || 0,
```
Good question! `marker` is guaranteed to exist, the way we structure this loop, and if marker exists, it will always have a `.dataset` property due through the standard DOM apis. Since `marker.dataset` will exist, the worst case scenario is `.long` evaluates to `undefined`, so `marker.dataset.long` will always equal `marker?.dataset?.long` — I felt that the `?.` therefore was potentially confusing

kadamwhite committed:
Merge branch 'main' into 1112-map-style-variant

kadamwhite committed:
Merge branch 'main' into 1112-map-style-variant

kadamwhite committed:
1112: Output projection attribute on frontend render

kadamwhite committed:
Add cachebusting for plugin CSS files

kadamwhite committed:
1112: Use grid to lay out side-by-side map variant

kadamwhite committed:
1112: Reconcile grid styles between new and old layouts

Adjust positioning to make the annual report maps continue to display properly
Set positioning (per HM recommendation, slight deviation from design) on side-by-side layout

kadamwhite committed:
1112: Allow map-level background color to be set

kadamwhite committed:
1112: Fix sidebar column margins

Sets things properly if map-level bg color is used,
but allows .alignwide groups to be nested in markers
if a custom bg color is needed

kadamwhite committed:
1112: Finesse styles for new side-by-side map variant content

kadamwhite committed:
1112: Once again fix the bug where projection isnt persisted to the frontend

kadamwhite committed:
1112: Add side-by-side map pattern

kadamwhite committed:
1112: Fix in-editor link color issue

kadamwhite commented:
Items hit during demo
- Deleting and re-adding map to page caused an error about the `map` container
- If you "duplicate" a Marker, it will use the same ID as a prior marker and won't show up properly on the map

kadamwhite committed:
1112: Remove dead code from marker block

kadamwhite committed:
1112: Fix issue where duplicating a marker block breaks the carousel

kadamwhite committed:
1112: Move arrows back into content column per Sara feedback

joeleenk commented on inc/assets.php:
```diff
@@ -220,3 +221,33 @@ function enqueue_frontend_styles() : void {
+ * @return string Filtered URL.
+ */
+function cachebust_stylesheets( string $src ): string {
+	$plugin_base_url = trailingslashit( plugin_dir_url( __DIR__ ) );
+	if ( strpos( $src, $plugin_base_url ) !== 0 ) {
+		return $src;
+	}
+
+	$file_relative_path = preg_replace( '/\?.*$/', '', str_replace( $plugin_base_url, '', $src ) );
+	$file_path          = plugin_dir_path( __DIR__ ) . $file_relative_path;
+	if ( ! is_readable( $file_path ) ) {
+		return $src;
+	}
+
+	$version = hash_file( 'crc32', $file_path ) ?: filemtime( $file_path );
```
These could use some comments to make it more quickly obvious what you're checking for.

joeleenk commented on src/blocks/map/style.scss:
```diff
@@ -106,3 +134,131 @@
+
+// These general non-layout styles are sandboxed to the side-by-side mode
+// as of July 2025. They could be expanded to apply to all maps if it is
+// acceptable to retroactively change design on e.g. old annual reports.
```
We've updated a lot of the other styles on older annual reports. The line has been colors or styles specific to that report stay, anything like fonts, spacing, site width that should be site-wide get updated. I'm not sure where these fall, but it seems like changes would be ok, as long as everything still works correctly, and has sufficient color contrast, etc.

kadamwhite replied:
Agree, but think we can maybe tackle that next time we work on the annual report. It was getting quite finicky right now :/

joeleenk commented on src/blocks/map/style.scss:
```diff
@@ -106,3 +134,131 @@
+		display: none;
+	}
+
+	.carousel-buttons {
```
I didn't dig into why (I can if you need me to), but the button position is weird on wider screens.

https://github.com/user-attachments/assets/9e8c787f-7ac3-4d67-a51c-75f4e28aa5db

kadamwhite replied:
Should be fixed with the revert to the in-sidebar button positions.

joeleenk reviewed (commented):
This mostly all looks great - the only blocker I see is the button positioning on wide screens.

kadamwhite committed:
Merge branch 'main' into 1112-map-style-variant

kadamwhite committed:
1112: Reposition arrows as per design direction

kadamwhite committed:
1112: Fix nested group bg color and CLS issues

- Set negative margins appropriately on nested groups to ensure that any
  alignfull groups with bg colors take up the whole height of the slide area.
- Adjust methodology for hiding slides on load-in to prevent CLS/flash
  of overlapping slides.

kadamwhite committed:
Document hashing approach for CSS files

joeleenk committed:
1112: Fix width for alignfull map

joeleenk reviewed (approved):
🚀
